python/cugraph/cugraph/dask/sampling/uniform_neighbor_sample.py

```diff
@@ -184,6 +189,13 @@ def uniform_neighbor_sample(
             df['hop_id']: dask_cudf.Series
                 Contains the hop ids from the sampling result
     """
+    if _multiple_clients:
+        # Distributed centralized lock to allow
+        # two disconnected processes (clients) to coordinate a lock
+        # https://docs.dask.org/en/stable/futures.html?highlight=lock#distributed.Lock
+        lock = Lock("sampling_mg_lock")
+        lock.acquire(timeout=1)
```

Comment:
If you don't get the lock and the timeout expires, should we do something? In this case, if the timeout expires it will run sampling anyway - is that okay?

Reply:
I think we can increase the timeout further , i think not running sampling will be problematic. How about we timeout at `50s` ? I think that should be enough .  The max i have seen it is 4s for a few million nodes at 1B scale. 

It is obviously a heuristic.

Reply:
Raised this with the last commit to 100.